src/doc/src/reference/semver.md

```diff
@@ -1114,6 +1115,87 @@ Mitigation strategies:
   optionally enables `std` support, and when the feature is off, the library
   can be used in a `no_std` environment.
 
+<a id="attr-adding-non-exhaustive"></a>
+### Major: adding `non_exhaustive` to an existing enum, variant, or struct with no private fields
+
+Making items [`#[non_exhaustive]`][non_exhaustive] changes how they may
+be used outside the crate where they are defined:
+- Non-exhaustive structs and enum variants cannot be constructed
+  using [struct literal] syntax, including [functional update syntax].
+- Pattern matching on non-exhaustive enums always requires
+  a wildcard (`_`) arm.
+
```

Comment:
BTW, a new rule is being added in 1.65 that non-exhaustive enums cannot be casted with `as` (see https://github.com/rust-lang/reference/pull/1249).

I'm reluctant to try to repeat the rules in this document since it will be difficult to keep in sync, or there may be subtle differences. However, this is useful information so it should probably be ok to repeat them here.